src/firefly/test/edu/caltech/ipac/astro/IpacTableReaderTest.java

```diff
@@ -0,0 +1,801 @@
+
+    @Test
+    /**
+     * This test calls the method IpacTableReader.readIpacTable to read IPACTable5_duplicatedCol.tbl
+     * which has duplicate column names.
+     */
+    public void testDuplicateColumn() throws IpacTableException, ClassNotFoundException, java.lang.NullPointerException {
+
+        /*
+        \catalog1 = 'Sample Catalog1'
+        \catalog2 = 'Sample Catalog2'
+        \ Comment1
+        \ Comment2
+        |   ra      |    ra     |   n_obs  |    V     |   SpType   |
+        |   double  |    double |   int    |   real   |   char     |
+        |   deg     |    deg    |          |   mag    |            |
+        |   null    |    null   |   null   |   null   |   null     |
+          165.466279  -34.704730      5       11.27       K6Ve
+          123.4       5.67            9       8.9         K6Ve-1
+         */
+
+        //Read a table with duplicated column names
+        String tableName = "IPACTable5_duplicatedCol.tbl";
+        File inFile = new File(FileLoader.getDataPath(IpacTableReaderTest.class) + tableName);
+
+        try{
+            dataGroup = IpacTableReader.readIpacTable(inFile, onlyColumns, useFloatsForDoubles, catName, isHeadersOnlyAllow);
+        } catch (IpacTableException e){
+            System.out.println("IpacTableException from the table " + tableName + ":" + e.getMessage());
+        }
+
+        DataType[] dataTypes = dataGroup.getDataDefinitions();
+        List<String> colTitle = new ArrayList<String>();
+        for (int i = 0; i < dataTypes.length; i++) {
+            colTitle.add(dataTypes[i].getDefaultTitle().toString());
+        }
+
+        System.out.println("NO IpacTableException from the table " + tableName + " which has duplicate column names: " + colTitle);
+
+        //Test result:
+        //The duplicate columns are not detected. A dataGroup is generated with those duplicate columns!
+        //Action item: Issue a ticket!
+
+    }
+
+    @Test
+    /**
+     * This test calls the method IpacTableReader.readIpacTable to read IPACTable6_nodata.tbl which has no data.
+     */
+    public void testNoData() throws IpacTableException, ClassNotFoundException, java.lang.NullPointerException {
+
+        /*
+        \catalog1 = 'Sample Catalog1'
+        \catalog2 = 'Sample Catalog2'
+        \ Comment1
+        \ Comment2
+        |   ra      |    dec    |   n_obs  |    V     |   SpType   |
+        |   double  |    double |   int    |   real   |   char     |
+        |   deg     |    deg    |          |   mag    |            |
+        |   null    |    null   |   null   |   null   |   null     |
+         */
+
+        //Read a table without data
+        String tableName = "IPACTable6_nodata.tbl";
+        File inFile = new File(FileLoader.getDataPath(IpacTableReaderTest.class) + tableName);
+
+        try{
+            dataGroup = IpacTableReader.readIpacTable(inFile, onlyColumns, useFloatsForDoubles, catName, isHeadersOnlyAllow);
+        } catch (IpacTableException e){
```

Comment:
This is the only Exception the IpacTableReader has and the one to be tested.  The correct way to test it is using
     try{
            dataGroup = IpacTableReader.readIpacTable(inFile, onlyColumns, useFloatsForDoubles, catName,  isHeadersOnlyAllow);
            Assert.fail("If no data error is not caught");
        } catch (IpacTableException e){
            System.out.println("IpacTableException from the table " + tableName + ": "  + e.getMessage());
        }